components/layout_2020/flow/float.rs

```diff
@@ -821,37 +821,71 @@ impl SequentialLayoutState {
         self.current_margin = CollapsedMargin::zero();
     }
 
-    /// Returns the amount of clearance that a block with the given `clear` value at the current
-    /// `bfc_relative_block_position` (with top margin included in `current_margin` if applicable)
-    /// needs to have.
+    /// Returns the amount of clearance (if any) that a block with the given `clear` value
+    /// needs to have at `current_block_position_including_margins()`.
+    /// The top margin must not be included in `current_margin`, pass it as `next_margin` instead.
+    /// The reason is that the presence of clearance must prevent `current_margin` and `next_margin`
+    /// from collapsing.
     ///
     /// https://www.w3.org/TR/2011/REC-CSS2-20110607/visuren.html#flow-control
-    pub(crate) fn calculate_clearance(&self, clear_side: ClearSide) -> Option<Length> {
+    pub(crate) fn calculate_clearance(
+        &self,
+        clear_side: ClearSide,
+        next_margin: &CollapsedMargin,
```

Comment:
Naming nit: How about `block_start_margin_of_clearing_element`? I know it's a mouthful, but `next_margin` is a bit ambiguous.

Reply:
Renamed to `block_start_margin`